packages/modeling/src/color/color.js

```diff
@@ -1,10 +1,37 @@
 const flatten = require('../utils/flatten')
 
+const geom2 = require('../geometry/geom2')
+const geom3 = require('../geometry/geom3')
+const path2 = require('../geometry/path2')
+const poly3 = require('../geometry/poly3')
+
+const colorGeom2 = (color, object) => {
+  const newgeom2 = geom2.create(geom2.toSides(object))
+  newgeom2.color = color
+  return newgeom2
+}
+
+const colorGeom3 = (color, object) => {
+  const newgeom3 = geom3.create(geom3.toPolygons(object))
+  newgeom3.color = color
+  return newgeom3
+}
+
+const colorPath2 = (color, object) => {
+  const newpath2 = path2.create(path2.toPoints(object))
+  newpath2.color = color
+  return newpath2
+}
+
+const colorPoly3 = (color, object) => {
+  object.color = color
```

Comment:
this is mutating the original still , so the same logic as above would work on the otherhand :)

Reply:
hmm but you want a new geometry containing the **new** polygon with a different color (immutable) right ? 
Ie, if you want to color an individual polygon (like in this case), it should return a new polygon imho

Reply:
Ah... colors are just an attribute at the geometry level. No more. We decided to support this long back.

`let newgeom = color([1,1,1,1], circle()) // geom2.color =[…]`

So, adding color to a polygon is a special case.

Reply:
@kaosat-dev I'm fine changing colorPoly3 to return a new polygon. Do you want this changed?

Reply:
Yes please, I think it makes sense to keep it all coherent (I agree that performance wise it is not the best, but it is an edge case, and immutability has pros & cons)

Reply:
Done